client/state/happychat/middleware.js

```diff
@@ -113,47 +103,29 @@ export const connectChat = ( connection, { getState, dispatch } ) => {
 		return;
 	}
 
+	const url = config( 'happychat_url' );
+
 	const user = getCurrentUser( state );
 	const locale = getCurrentUserLocale( state );
 	let groups = getGroups( state );
-
-	// update the chat locale and groups when happychat is initialized
 	const selectedSite = getHelpSelectedSite( state );
 	if ( selectedSite && selectedSite.ID ) {
 		groups = getGroups( state, selectedSite.ID );
 	}
 
-	// Notify that a new connection is being established
-	dispatch( setConnecting() );
-
-	debug( 'opening with chat locale', locale );
-
-	// Before establishing a connection, set up connection handlers
-	connection
-		.on( 'connected', () => {
-			dispatch( setConnected() );
-
-			// TODO: There's no need to dispatch a separate action to request a transcript.
-			// The HAPPYCHAT_CONNECTED action should have its own middleware handler that does this.
-			dispatch( requestChatTranscript() );
-		} )
-		.on( 'disconnect', reason => dispatch( setDisconnected( reason ) ) )
-		.on( 'reconnecting', () => dispatch( setReconnecting() ) )
-		.on( 'message', event => dispatch( receiveChatEvent( event ) ) )
-		.on( 'status', status => dispatch( setHappychatChatStatus( status ) ) )
-		.on( 'accept', accept => dispatch( setHappychatAvailable( accept ) ) );
+	const happychatUser = {
+		signer_user_id: user.ID,
+		locale,
+		groups,
+	};
 
-	// create new session id and get signed identity data for authenticating
 	return startSession()
 		.then( ( { session_id, geo_location } ) => {
-			if ( geo_location && geo_location.country_long && geo_location.city ) {
-				dispatch( setGeoLocation( geo_location ) );
-			}
-
+			happychatUser.geo_location = geo_location;
 			return sign( { user, session_id } );
 		} )
-		.then( ( { jwt } ) => connection.open( user.ID, jwt, locale, groups ) )
-		.catch( e => debug( 'failed to start happychat session', e, e.stack ) );
+		.then( ( { jwt } ) => connection.init( url, dispatch, { jwt, ...happychatUser } ) )
+		.catch( e => debug( 'failed to start Happychat session ', e, e.stack ) );
```

Comment:
Do we need the extra space at the end of "Happychat session "? that'll probably put two spaces when it prints

Reply:
Fixed in 4646789